packages/transform/src/template-to-typescript.ts

```diff
@@ -41,6 +41,14 @@ export function templateToTypescript(
   return mapTemplateContents(template, (ast, { emit, record, rangeForLine, rangeForNode }) => {
     let scope = new ScopeStack([]);
 
+    const firstNode = ast.body[0];
+    if (firstNode?.type === 'MustacheCommentStatement') {
+      let text = firstNode.value.trim();
+      if (/^@glint-nocheck/i.test(text)) {
+        return;
```

Comment:
I think we may want to consider implementing this more like `@glint-ignore` and `@glint-expect-error` work, where we [capture the existence of the directive as part of transforming the module](https://github.com/typed-ember/glint/blob/main/packages/transform/src/template-to-typescript.ts#L151-L167) and then [apply it after the fact](https://github.com/typed-ember/glint/blob/main/packages/core/src/common/transform-manager.ts#L268-L273) when transforming diagnostics on the way out.

The advantage to that approach is that it still provides editor support (hover info, go-to-definition, etc) in templates with `@glint-nocheck`, whereas short-circuiting the transformation like this will opt out of having Glint do anything with the template at all.